worker/codalabworker/worker.py

```diff
@@ -385,3 +390,226 @@ def _upgrade(self):
+
+        # 3) Set up container
+        if run_state.resources['request_network']:
+            docker_network = self.docker_network_external_name
+        else:
+            docker_network = self.docker_network_internal_name
+
+        cpuset, gpuset = self._assign_cpu_and_gpu_sets(
+                run_state.resources['request_cpus'], run_state.resources['request_gpus'])
+
+        # 4) Start container
+        container_id = self._docker.start_container(
+            run_state.bundle_path, bundle_uuid, run_state.bundle['command'],
+            run_state.resources['docker_image'], docker_network, dependencies,
+            cpuset, gpuset, run_state.resources['request_memory']
+        )
+
+        digest = self._docker.get_image_repo_digest(run_state.resources['docker_image'])
+        #self._image_manager.touch_image(digest)
+
+        return run_state._replace(stage=RunStage.RUNNING, run_status='Running',
+            container_id=container_id, docker_image=digest, cpuset=cpuset, gpuset=gpuset)
+
+    def _transition_run_state_from_RUNNING(self, run_state):
+        def check_and_report_finished(run_state):
+            try:
+                finished, exitcode, failure_msg = self._docker.check_finished(run_state.container_id)
+            except DockerException:
+                traceback.print_exc()
+                finished, exitcode, failure_msg = False, None, None
+            return dict(finished=finished, exitcode=exitcode, failure_message=failure_msg)
+
+        bundle_uuid = run_state.bundle['uuid']
+
+        new_info = check_and_report_finished(run_state)
+        run_state.info.update(new_info)
+        run_state = run_state._replace(info=run_state.info)
+        if run_state.info['finished']:
+            logger.debug('Finished run with UUID %s, exitcode %s, failure_message %s',
+                 bundle_uuid, run_state.info['exitcode'], run_state.info['failure_message'])
+            return run_state._replace(stage=RunStage.UPLOADING_RESULTS, run_status='Uploading results')
+        else:
+            return run_state
+
+    def _transition_run_state_from_UPLOADING_RESULTS(self, run_state):
+        def upload_results():
+            try:
+                # Delete the container.
+                if run_state.container_id is not None:
+                    while True:
+                        try:
+                            finished, _, _ = self._docker.check_finished(run_state.container_id)
+                            if finished:
+                                self._docker.delete_container(run_state.container_id)
+                                break
+                        except DockerException:
+                            traceback.print_exc()
+                            time.sleep(1)
+
+                # Clean-up dependencies.
+                for dep in run_state.bundle['dependencies']:
+                    #self._dependency_manager.remove_dependency(
```

Comment:
Why are these commented out?

Reply:
because this method has not yet been properly implemented